source/agora/node/Ledger.d

```diff
@@ -342,6 +370,13 @@ public class Ledger
 
         if (this.onAcceptedBlock !is null)
             this.onAcceptedBlock(block, validators_changed);
+
+        if (updateValidatorIndexMaps)
```

Comment:
I think this should be called before `onAcceptedBlock`, because that routine can cause a fiber switch.

Reply:
I am running all tests now with this change to check it doesn't break anything.

Reply:
Yes they pass will swap the order.